### ISSUE TITLE
Issue with Duration Parsing in Command API

### ISSUE THREAD

**Issue by AC-Lover** [Sonarr/Sonarr#7461]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

When calling the API endpoint `GET /api/v3/command/458143`, an error occurs during the parsing of the `Duration` column. The error message is:  

```
System.Data.DataException: Error parsing column 8 (Duration=00:00:01.7947962 - String)  
 ---> System.InvalidCastException: Invalid cast from 'System.String' to 'System.TimeSpan'.
```

The value of `Duration` for the command with ID `458143` is `00:00:01.7947962`, which seems to include microseconds. This format may not be compatible with `System.TimeSpan`, leading to the parsing failure.

### Expected Behavior

The `GET /api/v3/command/458143` API call should return the command's data successfully without any errors. The `Duration` field should either be parsed correctly or follow a supported format, such as `hh:mm:ss` or `hh:mm:ss.fff`.

### Environment

```markdown
- OS: Debian GNU/Linux 11 (bullseye)
- Sonarr: 4.0.11.2680
- Docker Install: No
- Using Reverse Proxy: Yes
- Browser: -
- Database: Sqlite 3.34.1
```

### What branch are you running?

Main

### Trace Logs?

2024-12-07 09:00:56.7|Trace|Http|Req: 1023 [GET] /sonarr/api/v3/command/458143?apikey=(removed) (from 10.106.1.254 curl/7.86.0)
2024-12-07 09:00:56.7|Debug|Sonarr.Http.Authentication.ApiKeyAuthenticationHandler|AuthenticationScheme: API was successfully authenticated.
2024-12-07 09:00:56.8|Trace|SonarrErrorPipeline|Handling Exception
2024-12-07 09:00:56.8|Fatal|SonarrErrorPipeline|Request Failed. GET /api/v3/command/458143

[v4.0.11.2680] System.Data.DataException: Error parsing column 8 (Duration=00:00:01.7947962 - String)
 ---> System.InvalidCastException: Invalid cast from 'System.String' to 'System.TimeSpan'.
   at System.Convert.DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
   at System.String.System.IConvertible.ToType(Type type, IFormatProvider provider)
   at Deserialize94159dfa-bbc4-46db-9ed3-a4820d7c6b45(IDataReader )
   --- End of inner exception stack trace ---
   at Dapper.SqlMapper.ThrowDataException(Exception ex, Int32 index, IDataReader reader, Object value) in /_/Dapper/SqlMapper.cs:line 3706
   at Deserialize94159dfa-bbc4-46db-9ed3-a4820d7c6b45(IDataReader )
   at Dapper.SqlMapper.QueryImpl[T](IDbConnection cnn, CommandDefinition command, Type effectiveType)+MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Dapper.SqlMapper.Query[T](IDbConnection cnn, String sql, Object param, IDbTransaction transaction, Boolean buffered, Nullable`1 commandTimeout, Nullable`1 commandType)
   at NzbDrone.Core.Datastore.SqlMapperExtensions.Query[T](IDatabase db, String sql, Object param)
   at NzbDrone.Core.Datastore.BasicRepository`1.Query(SqlBuilder builder)
   at NzbDrone.Core.Datastore.BasicRepository`1.Find(Int32 id)
   at NzbDrone.Core.Datastore.BasicRepository`1.Get(Int32 id)
   at NzbDrone.Core.Messaging.Commands.CommandQueueManager.Get(Int32 id)
   at Sonarr.Api.V3.Commands.CommandController.GetResourceById(Int32 id)
   at Sonarr.Http.REST.RestController`1.GetResourceByIdWithErrorHandler(Int32 id)
   at lambda_method243(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Sonarr.Http.Middleware.BufferingMiddleware.InvokeAsync(HttpContext context)
   at Sonarr.Http.Middleware.IfModifiedMiddleware.InvokeAsync(HttpContext context)
   at Sonarr.Http.Middleware.CacheHeaderMiddleware.InvokeAsync(HttpContext context)
   at Sonarr.Http.Middleware.StartingUpMiddleware.InvokeAsync(HttpContext context)
   at Sonarr.Http.Middleware.UrlBaseMiddleware.InvokeAsync(HttpContext context)
   at Sonarr.Http.Middleware.VersionMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)

2024-12-07 09:00:56.8|Trace|Http|Res: 1023 [GET] /sonarr/api/v3/command/458143?apikey=(removed) 500.InternalServerError (33 ms)

**Comment by stevietv** [comment]
I presume you didnt see any of responses on Discord before you raised this issue then?

00:00:01.794796 is a valid value to be able to cast to a timespan
Maybe you want to try updating your sqlite to a more recent version. otherwise, you could edit your sonarr db and drop the contents on the commands table

**Comment by rhass** [comment]
For what it's worth, I had this issue today when upgrading from Sonarr v3. However, I tried installing the current devel version which has this fix and had the same problem. I had a snapshot to revert back quickly, and it turned out the cause of the issue was torrents which were still in the queue. I was able to fix the issue by reverting the snapshot of the system to before upgrading, clearing the queue list, then going through the upgrade processes.
